Sources/swift-doc/Subcommands/Generate.swift

```diff
@@ -58,14 +58,17 @@ extension SwiftDoc {
 
         var pages: [String: Page] = [:]
 
+        var declaredTypeNames: [String] = []
```

Comment:
Is there any reason why we're collecting them here, rather than querying `module.interface` for symbols for which `symbol is Type || symbol is TypeAlias`?

Reply:
There was a reason which I don't remember exactly anymore. I think filtering a second time put quite some additional time in large code bases (e.g. bigger iOS apps and not only small libraries).

However, I also think that this spreads into unrelated logic and makes the code harder to understand. As we don't have too many performance improvements yet, maybe we can not spread the logic, filter a second time and then put some focus on performance in the future.

Reply:
Okay, forget about all of this.

It's rather easy and definitely the better approach to use all information which we have available on `Interface`, e.g. `symbolsGroupedByQualifiedName`. This will not only remove the spreading of the logic, it will also enable better detection of external symbols.

I'll pick this up later this week.